Code/Framework/AzFramework/Platform/Windows/AzFramework/Input/Devices/Mouse/InputDeviceMouse_Windows.h

```diff
@@ -0,0 +1,104 @@
+        SystemCursorState GetSystemCursorState() const override;
+
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        //! \ref AzFramework::InputDeviceMouse::Implementation::SetSystemCursorPositionNormalized
+        void SetSystemCursorPositionNormalized(AZ::Vector2 positionNormalized) override;
+
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        //! \ref AzFramework::InputDeviceMouse::Implementation::GetSystemCursorPositionNormalized
+        AZ::Vector2 GetSystemCursorPositionNormalized() const override;
+
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        //! \ref AzFramework::InputDeviceMouse::Implementation::TickInputDevice
+        void TickInputDevice() override;
+
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        //! \ref AzFramework::RawInputNotificationsWindows::OnRawInputEvent
+        void OnRawInputEvent(const RAWINPUT& rawInput) override;
+
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        //! Refresh system cursor clipping constraint
+        void RefreshSystemCursorClippingConstraint();
+
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        //! Refresh system cursor viibility
+        void RefreshSystemCursorVisibility();
+
+        //! Return the size of the client window constrained to the desktop work area.
+        RECT GetConstrainedClientRect(HWND focusWindow) const;
+
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        //! The current system cursor state
+        SystemCursorState m_systemCursorState;
+
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        //! Does the window attached to the input (main) thread's message queue have focus?
+        // https://msdn.microsoft.com/en-us/library/windows/desktop/ms646294(v=vs.85).aspx
+        bool m_hasFocus;
+
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        //! The client rect of the window obtained the last time this input device was ticked.
+        RECT m_lastClientRect;
+
+        //! The flags sent with the last received MOUSE_MOVE_RELATIVE or MOUSE_MOVE_ABSOLUTE event.
+        USHORT m_lastMouseMoveEventFlags;
+
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        //! The last absolute mouse position, reported by a MOUSE_MOVE_ABSOLUTE raw input API event,
+        //! which will more than likely only ever be received when running a remote desktop session.
+        AZ::Vector2 m_lastMouseMoveEventAbsolutePosition;
```

Comment:
Can you add curly braces on these variables to zero intialize them.

That will make the WinAPI RECT default to 0 for it's left, top, right and bottom members.

It will also make the AZ::Vector2 x and y members default to zero as well
```suggestion
        //! The current system cursor state
        SystemCursorState m_systemCursorState{};

        ////////////////////////////////////////////////////////////////////////////////////////////
        //! Does the window attached to the input (main) thread's message queue have focus?
        // https://msdn.microsoft.com/en-us/library/windows/desktop/ms646294(v=vs.85).aspx
        bool m_hasFocus{};

        ////////////////////////////////////////////////////////////////////////////////////////////
        //! The client rect of the window obtained the last time this input device was ticked.
        RECT m_lastClientRect{};

        //! The flags sent with the last received MOUSE_MOVE_RELATIVE or MOUSE_MOVE_ABSOLUTE event.
        USHORT m_lastMouseMoveEventFlags{};

        ////////////////////////////////////////////////////////////////////////////////////////////
        //! The last absolute mouse position, reported by a MOUSE_MOVE_ABSOLUTE raw input API event,
        //! which will more than likely only ever be received when running a remote desktop session.
        AZ::Vector2 m_lastMouseMoveEventAbsolutePosition{};
```